src/spinnaker_ros_interface_main.cc

```diff
@@ -265,8 +274,12 @@ void CaptureLoop(CameraPtr pCam) {
                   << pResultImage->GetImageStatus() 
                   << endl;
       } else {
-        image.header.stamp.fromSec(
-            1e-9 * static_cast<double>(pResultImage->GetTimeStamp()));
+        // Executes on the first frame only and gives us the FLIR camera's clock offset
+        if( net_offset == 0.0 )
+        {
+          net_offset = ros_time_offset.sec - 1e-9 * ( static_cast<double>(pResultImage->GetTimeStamp()) - ros_time_offset.nsec );
```

Comment:
Measure the difference between ros time and FLIR camera time here, and just save that difference.